python/lsst/dm/OCPS/ocps_csc.py

```diff
@@ -0,0 +1,230 @@
+            job_id = result.json().job_id
+            status_url = f"{self.config.url}/{job_id}"
+        else:
+            if data.pipeline not in ("true.yaml", "false.yaml", "fault.yaml"):
+                raise salobj.ExpectedError(f"Unknown pipeline: {data.pipeline}")
+            job_id = f"{data.pipeline}-{time.time()}"
+            status_url = f"ocps://{job_id}"
+            self.log.info(f"Simulated POST result: {status_url}")
+            self.simulated_jobs.add(job_id)
+
+        payload = json.dumps(dict(job_id=job_id))
+        self.cmd_execute.ack_in_progress(data, timeout=600.0, result=payload)
+        self.log.info(f"Ack in progress: {payload}")
+        self.log.info(f"Starting async wait: {status_url}")
+
+        while True:
+            if status_url.startswith("ocps://"):
+                await asyncio.sleep(abs(random.normalvariate(10, 4)))
+                # simulation mode
+                self.log.info(f"Simulating result for {job_id}")
+                if job_id not in self.simulated_jobs:
+                    raise salobj.ExpectedError(f"No such job id: {job_id}")
+                self.simulated_jobs.remove(job_id)
+                if job_id.startswith("true.yaml-"):
+                    payload = json.dumps(dict(result=True))
+                    self.evt_job_result.set_put(
+                        job_id=job_id, exit_code=0, result=payload
+                    )
+                elif job_id.startswith("false.yaml-"):
+                    payload = json.dumps(dict(result=False))
+                    self.evt_job_result.set_put(
+                        job_id=job_id, exit_code=0, result=payload
+                    )
+                elif job_id.startswith("fault.yaml-"):
+                    self.fault(
+                        code=2, report="404 Simulation cannot contact execution service"
+                    )
+                    raise salobj.ExpectedError("Failed to connect (simulated)")
+                else:
+                    raise salobj.ExpectedError(f"Unknown pipeline: {job_id}")
+                return
+
+            self.log.info(f"GET: {status_url}")
+            result = self.connection.get(status_url)
+            result.raise_for_status()
+            response = result.json()
+            if response.status != "ok":
+                raise salobj.ExpectedError(f"GET {status_url} failed: {response}")
+            job_id = status_url[-status_url.rindex("/") :]
+            if response.job.state == "running":
+                self.log.debug(f"{status_url} sleeping for {self.config.poll_interval}")
+                await asyncio.sleep(self.config.poll_interval)
+            else:
+                self.log.info(f"{status_url} result: {response}")
+                self.evt_job_result.set_put(
+                    job_id=job_id, exit_code=0, result=response.job
+                )
+                return
+
+    async def do_abort_job(self, data):
+        """Implement the ``abort_job`` command.
```

Comment:
Our standard for SAL topics is to use camelCase for topic and field names. So this should be do_abortJob and the job_id field should be jobId. It is unfortunate and makes a mess in python.

Reply:
This is already in ts_xml, unfortunately, and I'd rather not have to wait for another XML release.  The same goes for `data_query` in the `execute` command.  On the other hand, it's arguably better to fix this now before anyone has actually used it.

Reply:
That's fair. "pre-existing condition" is a valid argument.